examples/tcp-socket-listener-client/tcp_socket_listener.bal

```diff
@@ -7,42 +7,47 @@ import ballerina/socket;
+    resource function onConnect(socket:Caller caller) {
         log:printInfo("Client connected: " + caller.id);
     }
 
     // This resource is invoked once the content is received from the client.
-    resource function onReadReady(socket:Caller caller, byte[] content) {
-        // Create a new `ReadableByteChannel` using the newly received content.
-        io:ReadableByteChannel byteChannel =
+    resource function onReadReady(socket:Caller caller) {
+        var result = caller->read();
+        if (result is (byte[], int)) {
+            var (content, length) = result;
+            if (length > 0) {
+                // Create a new `ReadableByteChannel` using the newly received content.
+                io:ReadableByteChannel byteChannel =
                     io:createReadableChannel(content);
-        io:ReadableCharacterChannel characterChannel =
+                io:ReadableCharacterChannel characterChannel =
                     new io:ReadableCharacterChannel(byteChannel, "UTF-8");
-        var str = characterChannel.read(20);
-        if (str is string) {
-            string reply = untaint str + " back";
-            byte[] payloadByte = reply.toByteArray("UTF-8");
-            // Send reply to the `caller`.
-            var writeResult = caller->write(payloadByte);
-            if (writeResult is int) {
-                log:printInfo("Number of bytes written: " + writeResult);
+                var str = characterChannel.read(20);
+                if (str is string) {
+                    string reply = untaint str + " back";
+                    byte[] payloadByte = reply.toByteArray("UTF-8");
+                    // Send reply to the `caller`.
+                    var writeResult = caller->write(payloadByte);
+                    if (writeResult is int) {
+                        log:printInfo("Number of bytes written: " + writeResult);
+                    } else {
+                        log:printError("Unable to write the content",
+                            err = writeResult);
+                    }
+                } else {
+                    log:printError("Error while writing content to the caller",
+                        err = str);
+                }
             } else {
-                log:printError("Unable to write the content",
-                    err = writeResult);
+                log:printInfo("Client leaved: " + caller.remotePort);
             }
         } else {
-            log:printError("Error while writing content to the caller",
-                err = str);
+            io:println(result);
         }
     }
 
-    // This resource is invoked when the client departs.
-    resource function onClose(socket:Caller caller) {
-        log:printInfo("Client left: " + caller.id);
-    }
-
     // This resource is invoked for the error situation
-    // if it happens during the `onAccept`, `onReadReady`, and `onClose`.
+    // if it happens during the `onConnect` and `onReadReady`.
```

Comment:
This resource is invoked by the  `onConnect` and `onReadReady`functions in erroneous situations.